trin-core/src/utils/node_id.rs

```diff
@@ -0,0 +1,35 @@
+use crate::utils::bytes;
+use crate::utils::distance::xor_two_values;
+use discv5::enr::NodeId;
+
+/// Generate random NodeId based on bucket index target and a local node id.
+/// First we generate a random distance metric with leading zeroes based on the target bucket.
+/// Then we XOR the result distance with the local NodeId to get the random target NodeId
+pub fn generate_random_node_id(
+    target_bucket_idx: u8,
+    local_node_id: NodeId,
+) -> anyhow::Result<NodeId> {
```

Comment:
I was experimenting to get rid of `OverlayRequestError` in `overlay_service` but it is quite of refactoring because we are now passing `Result<T, E>` between the channels and `anyhow::Error` does not implement the `Clone` trait. Also, there are quite a few important PR in progress, so maybe we don't want to do such refactoring at this moment.